tests/integration/components/async-button-test.js

```diff
@@ -0,0 +1,25 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+const { run } = Ember;
+
+moduleForComponent('async-button', 'Integration | Component | async button', {
+  integration: true
+});
+
+test('button calls submit action', function(assert) {
+  assert.expect(1);
+  this.disabled = false;
+  this.on('saveAction', () => {
+    this.set('disabled', true);
+    assert.ok(true, 'this action is only called once');
+  });
+
+  this.render(hbs`{{async-button submitAction=(action "saveAction") disabled=disabled}}`);
+  this.$('input').click();
+
+  run(() => {
```

Comment:
We explicitly use `run` here to ensure that the passed in action is not called a second time once `disabled` is set to true.